sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1902,249 @@ def transform(self, X):
+        counted
+        - 2D list of indices : The given list of lists of features is counted,
+        but each list in the list of lists have its own set of counts
+
+    Attributes
+    ----------
+    count_cache_ : defaultdict(int)
+        The counts of each example learned during 'fit'
+
+    classes_ : list of (index, y) tuples
+        An enumerated set of all unique values 'y' can have
+
+    col_num_X_ : int
+        The number of columns of 'X' learned during 'fit'
+        We use this to compare to the number of columns of 'X'
+        during transform to make sure that the transformation is compatible
+
+    col_num_Y_ : int
+        The number of columns of 'y' learned during 'fit'
+        If 0, then the fit is not conditional on the y given
+
+    Examples
+    --------
+    Given a dataset with two features and four samples, we let the transformer
+    find the number of occurences of each data point in the dataset
+    Note how the first column duplicates the input data, the second column
+    corresponds to the count of ``y=0`` and the third column corresponds to
+    the count of ``y=1``.
+
+    >>> from sklearn.preprocessing.data import CountFeaturizer
+    >>> X = [[0], [0], [0], [0], [1], [1], [1], [1]]
+    >>> y = [0, 1, 1, 1, 0, 0, 0, 0]
+    >>> cf = CountFeaturizer()
+    >>> cf.fit_transform(X, y)      # doctest: +NORMALIZE_WHITESPACE
+    array([[ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.]])
+
+    See also
+    --------
+    https://blogs.technet.microsoft.com/machinelearning/2015/02/17/big-learning-made-easy-with-counts/      # noqa
+    https://msdn.microsoft.com/en-us/library/azure/dn913056.aspx
+    """
+
+    def __init__(self, inclusion='all'):
+        self.inclusion = inclusion
+
+    @staticmethod
+    def _check_params(inclusion=None):
+        if inclusion is None:
+            raise ValueError("Inclusion cannot be none")
+        if (inclusion is not None and (type(inclusion) == str and
+            inclusion != "all" and inclusion != "each") and
+                not CountFeaturizer._valid_data_type(inclusion)):
+            raise ValueError("Illegal data type in inclusion")
+
+    @staticmethod
+    def _valid_data_type(type_check):
+        return type(type_check) == np.ndarray or type(type_check) == list
+
+    def _get_inclusion_used(self):
+        if type(self.inclusion) == str and self.inclusion == "all":
+            return np.array([range(self.col_num_X_)])
+        elif type(self.inclusion) == str and self.inclusion == "each":
+            return np.array([[i] for i in range(self.col_num_X_)])
+        elif CountFeaturizer._valid_data_type(self.inclusion):
+            inclusion_used = np.array(self.inclusion)
+            if len(inclusion_used.shape) == 1:
```

Comment:
Hmm. I just made it so that 
```
        elif CountFeaturizer._valid_data_type(self.inclusion): # meaning that inclusion is a list or np.ndarray
            return self.inclusion
```
because of how I was using `inclusion_used`, it didn't make a difference whether it was a `np.ndarray` or a `list`